lib/src/query_snapshot_stream_manager.dart

```diff
@@ -131,6 +132,22 @@ class QuerySnapshotStreamManager {
         if (docChange != null) {
           documentsChange.add(docChange);
         }
+
+        /// When multiple new doc snapshots added for the current state, we need to fire multiple document changes
+        final newDocsChanges = docsCurrent.where((doc) {
```

Comment:
You're checking changes for documents that have appeared since. So if some document is in `docsPrior` but not `docsCurrent`, this logic won't catch the change:
```typescript
    // Remove in parallel.
    final op3 = db.collection('docs').doc('1').set(<String, dynamic>{
      'age': 36,
    }, SetOptions(merge: true));
    final op4 = db.collection('docs').doc('2').set(<String, dynamic>{
      'age': 40,
    }, SetOptions(merge: true));
    await Future.wait([op3, op4]);
```
Only emits `[DocumentChangeType.removed]`, not `[DocumentChangeType.removed, DocumentChangeType.removed]`.

I'll try and submit a PR for it.